src/XMonad/Operations.hs

```diff
@@ -491,6 +498,9 @@ readStateFile xmc = do
                            [(x, "")] -> Just x
                            _         -> Nothing
 
+    readStrict :: Handle -> IO String
```

Comment:
It would probably be much easier to just use [Data.ByteString.readFile](http://hackage.haskell.org/package/bytestring-0.10.8.2/docs/Data-ByteString.html#v:readFile), which is strict.